### PR TITLE
cloudapi: extend the API spec with /version and /openapi.json

### PR DESCRIPTION
**Pull request by msehnout**
These endpoints are useful for clients while exploring the API. They are
also required for deploying the service into clouddot.

This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change

### CODE REVIEW

**Review by teg — COMMENTED** [review]
Nice! Should wait for docs + tests + CI, but otherwise: looks great to me! @Gundersanne could you have a look to make sure this ends up consistent with what we do in image builder?

File: vendor/modules.txt [hunk 1]
```diff
@@ -100,6 +100,7 @@ github.com/dgrijalva/jwt-go
 # github.com/dimchansky/utfbom v1.1.0
 github.com/dimchansky/utfbom
 # github.com/getkin/kin-openapi v0.13.0
+## explicit
```
Comment (Gundersanne) [inline vendor/modules.txt:103]: What does this do?

**Review by Gundersanne — COMMENTED** [review]
Sorry for the late review, this looks fine! I have 1 question though. 

Oh and we should test both of these in `test/cases/api.sh`

File: vendor/modules.txt (hunk 1 above)
Comment (msehnout) [reply vendor/modules.txt:103]: I'm not sure. This is automatically generated but it is all around this file. Maybe another legacy stuff cause by Go import path vs. Go modules? https://golang.org/cmd/go/#hdr-Module_compatibility_and_semantic_versioning

File: osbuild-composer.spec [hunk 2]
```diff
@@ -46,6 +46,7 @@ BuildRequires:  golang(github.com/deepmap/oapi-codegen/pkg/codegen)
 BuildRequires:  golang(github.com/go-chi/chi)
 BuildRequires:  golang(github.com/google/uuid)
 BuildRequires:  golang(github.com/julienschmidt/httprouter)
+BuildRequires:  golang(github.com/getkin/kin-openapi)
```
Comment (ondrejbudai) [inline osbuild-composer.spec:49]: This should fix it:

```suggestion
BuildRequires:  golang(github.com/getkin/kin-openapi/openapi3)
```

**Review by teg — APPROVED** [review]
Looking great!

**Comment by teg** [comment]
Oh, CI needs to be retriggered. Schutzbot is not here.

**Comment by msehnout** [comment]
> Oh, CI needs to be retriggered. Schutzbot is not here.

Exactly. Let's wait for Schutzbot to finish.

**Comment by msehnout** [comment]
OSTree test failed on:
```
[2021-02-12T08:12:18.716Z] TASK [assert] ******************************************************************
[2021-02-12T08:12:18.716Z] fatal: [192.168.100.50]: FAILED! => {
[2021-02-12T08:12:18.716Z]     "assertion": "result_dmesg_error.stdout_lines | length == 2",
[2021-02-12T08:12:18.716Z]     "changed": false,
[2021-02-12T08:12:18.716Z]     "evaluated_to": false,
[2021-02-12T08:12:18.716Z]     "msg": "more or less error and failed log"
[2021-02-12T08:12:18.716Z] }
```
Fedora 32 Base failed on:
```
...
[2021-02-12T08:03:06.823Z] Errors during downloading metadata for repository '3':
[2021-02-12T08:03:06.824Z]   - Curl error (23): Failed writing received data to disk/application for http://mirror.genesisadaptive.com/fedora/updates/33/Modular/x86_64/repodata/be4d0918170edb9448dd73a9e69127782ee7685af456ca00d0d2a533dcf4960e-filelists.xml.zck [Failed writing body (0 != 322)]
[2021-02-12T08:03:06.824Z]   - Curl error (28): Timeout was reached for https://mirror.csis.ysu.edu/fedora/fedora/linux/updates/33/Modular/x86_64/repodata/be4d0918170edb9448dd73a9e69127782ee7685af456ca00d0d2a533dcf4960e-filelists.xml.zck [Operation too slow. Less than 1000 bytes/sec transferred the last 5 seconds]
[2021-02-12T08:03:06.824Z] === CONT  TestCrossArchDepsolve/fedora-33/x86_64/ami
[2021-02-12T08:03:06.824Z]     main_test.go:85: 
[2021-02-12T08:03:06.824Z]         	Error Trace:	main_test.go:85
[2021-02-12T08:03:06.824Z]         	Error:      	Received unexpected error:
[2021-02-12T08:03:06.824Z]         	            	DNF error occured: RepoError: Error occurred when setting up repo: Failed to download metadata for repo '3': Yum repo downloading error: Downloading error(s): repodata/be4d0918170edb9448dd73a9e69127782ee7685af456ca00d0d2a533dcf4960e-filelists.xml.zck - Download failed: Curl error (23): Failed writing received data to disk/application for http://mirror.genesisadaptive.com/fedora/updates/33/Modular/x86_64/repodata/be4d0918170edb9448dd73a9e69127782ee7685af456ca00d0d2a533dcf4960e-filelists.xml.zck [Failed writing body (0 != 322)]
[2021-02-12T08:03:06.824Z]         	Test:       	TestCrossArchDepsolve/fedora-33/x86_64/ami
...
[2021-02-12T08:03:55.307Z]             --- FAIL: TestCrossArchDepsolve/fedora-33/x86_64/ami (701.33s)
```

**Comment by msehnout** [comment]
I don't think the CI failures are related. The first one seems to complain about wrong number of errors during OSTree image boot and the second is just a slow mirror. Can you check it again @Gundersanne ? Mainly if you are ok with the way these endpoints are tests.

**Review by Gundersanne — APPROVED** [review]
lgtm :)

**Pull request merged**